packages/sdk/src/streamStateView_Space.ts

```diff
@@ -150,6 +131,62 @@ export class StreamStateView_Space extends StreamStateView_AbstractContent {
         }
     }
 
+    private decryptionInProgress: Promise<ChunkedMedia | undefined> | undefined
+    private decryptionQueue: Promise<void> = Promise.resolve()
+    private latestEncryptedImage: EncryptedData | undefined
+
+    public async getSpaceImage(): Promise<ChunkedMedia | undefined> {
+        // Take a snapshot of the current encrypted image
+        const currentEncryptedImage = this.encryptedSpaceImage
+
+        // If the current image is already being decrypted, return the ongoing decryption promise
+        if (this.decryptionInProgress && currentEncryptedImage === this.latestEncryptedImage) {
+            return this.decryptionInProgress
+        }
+
+        if (currentEncryptedImage) {
+            const spaceAddress = contractAddressFromSpaceId(this.streamId)
+            const context = spaceAddress.toLowerCase()
+
+            // Update the latest encrypted image
+            this.latestEncryptedImage = currentEncryptedImage
+
+            // Chain the decryption operations sequentially
+            this.decryptionQueue = this.decryptionQueue.then(async () => {
+                // Assign the decryption process to decryptionInProgress and return void to avoid ESLint warnings
+                this.decryptionInProgress = this.performDecryption(context, currentEncryptedImage)
+                return this.decryptionInProgress.then(() => undefined) // Ensure void return
+            })
+
+            await this.decryptionQueue
+
+            // Always return the last known good state of spaceImage, even if it wasn't updated
+            return this.spaceImage
```

Comment:
you can delete line 163 and 164, you don't need duplicate returns

Reply:
good eye. removed